format/stmt.go

```diff
@@ -47,8 +47,37 @@ func (p *printer) stmt(s stmt.Stmt) {
 		p.expr(s.Chan)
 		p.buf.WriteString("<-")
 		p.expr(s.Value)
+	case *stmt.Switch:
+		p.buf.WriteString("switch ")
+		if s.Init != nil {
+			p.stmt(s.Init)
+			if s.Cond != nil {
+				p.buf.WriteString("; ")
+			}
+		}
+		if s.Cond != nil {
+			p.expr(s.Cond)
+		}
+		p.buf.WriteString("{\n")
```

Comment:
You only want to print the newline if there are any cases. "select {}" is good form.

Reply:
done.